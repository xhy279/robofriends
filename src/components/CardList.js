import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
  // test ErrorBoundry
  // if (1) {
  //   throw new Error('dd');
  // }
  return ( 
    <div>
      { robots.map((i) => {
          return <Card key={i.id} id={i.id} name={i.name} email={i.email} />
      })}
    </div>
  );
};

export default CardList;