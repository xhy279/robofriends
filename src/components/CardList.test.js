import React from 'react';
import CardList from './CardList';
import { shallow } from 'enzyme'; 
import toJson from 'enzyme-to-json';

it('expect to match snapshot', () => {
  const mockRobots = [{
    id: 1,
    name: 'a',
    email: 'a@a.com',
  }];
  const wrapper = shallow(<CardList robots={mockRobots}/>);
  expect(toJson(wrapper)).toMatchSnapshot();
});