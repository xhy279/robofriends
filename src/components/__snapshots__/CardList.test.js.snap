// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to match snapshot 1`] = `
<div>
  <Card
    email="a@a.com"
    id={1}
    key="1"
    name="a"
  />
</div>
`;
